### PR TITLE
Mutually exclusive options

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Recurring events and Stripe payment should not be enabled at the same time, as there is no workflow put in place to make sense on how would that work, charging by series, by booking, rescheduling, cancelling a few occurrences, etc.

This PR makes sure each option on event type is mutually exclusive.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

 - Go to a Payment is enabled event type configuration and see recurring event option is gone. 
 - When Payment option is disabled, recurring event option should be in place.

### CODE REVIEW

File: apps/web/components/eventtype/RecurringEventController.tsx [hunk 1]
```diff
@@ -33,100 +40,105 @@ export default function RecurringEventController({
...
-                }
-                recurringEvent = formMethods.getValues("recurringEvent");
-              }}
-              type="checkbox"
-              className="text-primary-600  h-4 w-4 rounded border-gray-300"
-              defaultChecked={recurringEventDefined}
-              data-testid="recurring-event-check"
-            />
+    <>
+      {!paymentEnabled && (
```
Comment (leog) [inline apps/web/components/eventtype/RecurringEventController.tsx:44]: If payment enabled, don't show recurring event option

**Review by alannnc — APPROVED** [review]
Code is good and works, but in my opinion we should at least let the user know what happened with the change. e.g. If selected one or another don't disappear the other one, we could disabled it with an explanation in a tooltip.

**Comment by agustif** [comment]
Wouldn't be best to actually try and make recurring events be able to be paid too? If we're involving @Jaibles and doing design work, we might already go for the feature that would be best?

Guessing most of the use-cases around recurring events could be paid events anyways, without much data to support it

**Comment by PeerRich** [comment]
yeah we need to define features that block others – however the ultimate goal is to be as compatible as possible.
 
it would be great if we could do paid recurring meetings

**Comment by leog** [comment]
> it would be great if we could do paid recurring meetings

We can do it, it just needs definition on how it should work. Paying for the entire series of occurrences and refunding cancelled instances may be an option, or any other combination. We need @Jaibles!

**Review by alishaz-polymath — APPROVED** [review]
Code looks good! Nice work 
As soon as @Jaibles can confirm his vision on this, we could take the next steps. 🚀

**Review by agustif — APPROVED** [review]
Great fix for now, let's see how to improve later with new designs!

**Comment by emrysal** [comment]
Actually don't like this just yet; it's not to do with the code (some state can be replaced with RHF but that's besides this PR) - My first question is "Why can't I make an unpaid recurring event?" - if the answer is because there's no workflow for it the real fix is to make a workflow for it. Logically this should not be mutually exclusive in my mind. 😖 

But even so, I think the UI should be clear and this is not, my two cents is that we shouldn't merge as is as it could also break existing configuration.

**Comment by leog** [comment]
> Actually don't like this just yet; it's not to do with the code (some state can be replaced with RHF but that's besides this PR) - My first question is "Why can't I make an unpaid recurring event?" - if the answer is because there's no workflow for it the real fix is to make a workflow for it. Logically this should not be mutually exclusive in my mind. 😖
> 
> But even so, I think the UI should be clear and this is not, my two cents is that we shouldn't merge as is as it could also break existing configuration.

The thing is there are already some users noticing inconsistencies on the UI as the payment option is not there when it should due to recurring events configuration, this solves a bug for the time being, can be revisited later to improve it and support payment full on with recurring events.

**Comment by Jaibles** [comment]
While this is not the best solution for this - right now I can't see the payment option at all in my event type even though I have Stripe installed - so if this fixes that bug I think we need to get it out and do a very quick follow up to make payments work for recurring events also.

**Comment by leog** [comment]
Agreed. Would this work @PeerRich @Jaibles @emrysal:

No option selected:
Pasted_Image_16_5_22__13_53

Recurring event option selected:
Pasted_Image_16_5_22__13_51

Payment option selected:
Pasted_Image_16_5_22__13_54

**Comment by Jaibles** [comment]
That's better yea 👍 Can we do a copy update?

If recurring event is enabled:
- After: `Payments are not currently supported with Recurring Events`
- Before: `Recurring events doesn't support payments yet, disabling the option`

If payments are enabled:
- After: `Recurring events are not currently supported with Payments`
- Before: `Recurring events doesn't support payments yet, disabling the option`

**Pull request merged**